Rename Final Offset to Final Size (#2285)
quicwg/base-drafts: draft-ietf-quic-transport.md

endpoints sends CONNECTION_CLOSE.

## Stream Final Size {#final-size}

The final size is the amount of data that is transmitted on a stream.  For a
stream that is reset, the final size is carried explicitly in a RESET_STREAM
frame.  Otherwise, the final size is the offset plus the length of a STREAM
frame marked with a FIN flag, or 0 in the case of incoming unidirectional

> streams.

Author: I think that the current draft is incorrectly referring to incoming unidirectional streams; see #2286.

endpoints sends CONNECTION_CLOSE.

## Stream Final Size {#final-size}

The final size is the amount of flow control credit that is consumed by a

> stream.  Assuming that every contiguous byte on the stream was sent, the final

Reviewer: ```suggestion
stream.  Assuming that every contiguous byte on the stream was sent once, the final
```

> stream.  Assuming that every contiguous byte on the stream was sent, the final

Reviewer 2: ```suggestion
stream.  When every contiguous byte on the stream is sent, the final
```

size is the number of bytes sent.  More generally, this is one higher than the
largest byte offset sent on the stream.

For a
stream that is reset, the final size is carried explicitly in a RESET_STREAM

> frame.  Otherwise, the final size is the offset plus the length of a STREAM

Reviewer 3: It might be worth adding that for a stream that is reset, the final size might be smaller than the number of bytes sent on the stream.

Author: Isn't that a protocol violation triggering FINAL_OFFSET_ERROR (to be renamed to FINAL_SIZE_ERROR)? Quoting from the draft: _an endpoint received a RESET_STREAM frame containing a final offset that was lower than the maximum offset of data that was already received._

Reviewer 3: Yes, my suggestion was incorrect. Not sure what I was thinking.

endpoints sends CONNECTION_CLOSE.

## Stream Final Size {#final-size}

The final size is the amount of flow control credit that is consumed by a
stream.  Assuming that every contiguous byte on the stream was sent once, the
final size is the number of bytes sent.  More generally, this is one higher

> than the largest byte offset sent on the stream.

Reviewer: Too late, but there is missing a "if any".

Reviewer 4: Never too late.  But I'm not following you.

Reviewer: The final size can be zero. That is not one larger than largest sent.
